pkg/kubeapiserver/authorizer/config.go

```diff
@@ -95,7 +98,15 @@ func (config Config) New() (authorizer.Authorizer, authorizer.RuleResolver, erro
 				config.VersionedInformerFactory.Core().V1().PersistentVolumes(),
 				config.VersionedInformerFactory.Storage().V1().VolumeAttachments(),
 			)
-			nodeAuthorizer := node.NewAuthorizer(graph, nodeidentifier.NewDefaultNodeIdentifier(), bootstrappolicy.NodeRules())
+			nodeAuthorizer := withNotSyncedReason{
+				node.NewAuthorizer(graph, nodeidentifier.NewDefaultNodeIdentifier(), bootstrappolicy.NodeRules()),
+				[]cache.SharedInformer{
+					config.VersionedInformerFactory.Core().V1().Nodes().Informer(),
+					config.VersionedInformerFactory.Core().V1().Pods().Informer(),
+					config.VersionedInformerFactory.Core().V1().PersistentVolumes().Informer(),
+					config.VersionedInformerFactory.Storage().V1().VolumeAttachments().Informer(),
+				},
```

Comment:
how do we make sure the list of informers is consistent with what's used inside?

Reply:
just manually by looking 10 lines above. But this is the case everywhere where we have similar code. Better pattern welcome.